platform-sdk/swirlds-common/src/testFixtures/java/com/swirlds/common/test/fixtures/crypto/PreGeneratedX509Certs.java

```diff
@@ -0,0 +1,398 @@
+    //        generateCerts(numCerts, getRandom());
+    //    }
+
+    /**
+     * Retrieves a pre-generated X.509 certificate for signing purposes.
+     *
+     * @param nodeId the node ID
+     * @return the X.509 certificate
+     */
+    @Nullable
+    public static SerializableX509Certificate getSigCert(final long nodeId) {
+        if (sigCerts.isEmpty()) {
+            loadCerts();
+            if (sigCerts.isEmpty()) {
+                return null;
+            }
+        }
+        long index = nodeId % sigCerts.size();
+        return sigCerts.get(new NodeId(index));
+    }
+
+    /**
+     * Retrieves a pre-generated X.509 certificate for TLS agreement use.
+     *
+     * @param nodeId the node ID
+     * @return the X.509 certificate
+     */
+    public static SerializableX509Certificate getAgreeCert(final long nodeId) {
+        if (agreeCerts.isEmpty()) {
+            loadCerts();
+            if (agreeCerts.isEmpty()) {
+                return null;
+            }
+        }
+        long index = nodeId % agreeCerts.size();
+        return agreeCerts.get(new NodeId(index));
+    }
+
+    /**
+     * Loads pre-generated X.509 certificates from disk.
+     */
+    private static void loadCerts() {
+        if (sigCertsHexEncoded.isEmpty() || agreeCertsHexEncoded.isEmpty()) {
+            // certs need to be generated first.
+            return;
+        }
+
+        try {
+            long nodeId = 0;
+
+            for (final String sigCertHexString : sigCertsHexEncoded) {
+                byte[] sigCertBytes = CommonUtils.unhex(sigCertHexString);
+                SerializableDataInputStream sigIs =
+                        new SerializableDataInputStream(new ByteArrayInputStream(sigCertBytes));
+                SerializableX509Certificate sigCert = sigIs.readSerializable(false, SerializableX509Certificate::new);
+                sigCerts.put(new NodeId(nodeId), sigCert);
+                nodeId++;
+            }
+
+            nodeId = 0;
+            for (final String agreeCertHexString : agreeCertsHexEncoded) {
+                byte[] agreeCertBytes = CommonUtils.unhex(agreeCertHexString);
+                SerializableDataInputStream agreeIs =
+                        new SerializableDataInputStream(new ByteArrayInputStream(agreeCertBytes));
+                SerializableX509Certificate agreeCert =
+                        agreeIs.readSerializable(false, SerializableX509Certificate::new);
+                agreeCerts.put(new NodeId(nodeId), agreeCert);
+                nodeId++;
+            }
+        } catch (final Exception e) {
+            throw new IllegalStateException("critical failure in loading certificates", e);
+        }
+        if (sigCerts.isEmpty() || agreeCerts.isEmpty()) {
+            throw new IllegalStateException("critical failure in loading certificates, no certs found.");
+        }
+    }
+
+    private static final List<String> sigCertsHexEncoded = List.of(
```

Comment:
I think the encoded certs should be in a separate file

Reply:
There is precedent for doing this: [PreGeneratedPublicKeys](https://github.com/hashgraph/hedera-services/blob/develop/platform-sdk/swirlds-common/src/testFixtures/java/com/swirlds/common/test/fixtures/crypto/PreGeneratedPublicKeys.java)

I originally had the keys in a separate file in the `swirld-common/src/testFixtures/resources` directory.   The problem came in when using `RandomAddressBookGenerator` from other modules.  I can't use an absolute PATH to the files to read them because other people may have them in different locations on their machines.  I also cannot use a relative PATH because the working directly in which people start the unit tests may be different.  

I agree having them in a separate file would be better, but I am ignorant of how to make that work so that it is insensitive to the current working directory that people initiate unit tests from.

Reply:
I will see if I can get the class loader resources to work despite the directory of execution.

Reply:
So far my efforts have failed to load the files in the resources directory.  The `PreGeneratedX509Certs` file is in `swirlds-common`, but the unit tests that need to have it load the certs are running in `swirlds-platform-core`.  When performing `PreGeneratedX509Certs.class.getClassLoader()`, null is returned.   I believe `Thread.currentThread.getContextClassLoader()` works, but its package binding is not going to be to `swirlds-common`, but whatever project's unit tests are executing.

Reply:
our module structure is still quite messy, it seems that it needs some cleanup in order to accommodate this